unstructured/partition/auto.py

```diff
@@ -141,12 +141,12 @@ def partition(
     encoding: Optional[str] = None,
     paragraph_grouper: Optional[Callable[[str], str]] | Literal[False] = None,
     headers: dict[str, str] = {},
-    skip_infer_table_types: list[str] = [],
+    skip_infer_table_types: list[str] = ["pdf", "jpg", "png", "heic"],
```

Comment:
You should really never do this. Initializing mutable default parameters, such as `skip_infer_table_types`, with a mutable data type (e.g., list, dict, etc.) can lead to unexpected behavior. The default value is shared across all instances of the function call, meaning that any modification to this parameter within one function call will affect subsequent calls. This can introduce subtle bugs that are difficult to trace and debug. Instead, initialize the parameter with `None` and set the default value inside the function if it is `None`.

Reply:
Thanks for the suggestion @MalteHB ! Opened #3063 to address.